src/applications/disability-benefits/2346/containers/OrderHistory.jsx

```diff
@@ -0,0 +1,228 @@
+import ErrorableCheckbox from '@department-of-veterans-affairs/formation-react/ErrorableCheckbox';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { focusElement } from 'platform/utilities/ui';
+import {
+  getDLCAccessoriesData,
+  getDLCBatteryData,
+  // NOTE: Decision was to take socks out of MVP -@maharielrosario at 1/28/2020, 9:49:20 AM
+  // getDLCSocksData,
+  updateDLCData,
+} from '../actions';
+
+class OrderHistory extends Component {
+  // NOTE: Using React state for checkboxes -@maharielrosario at 1/8/2020, 7:47:21 PM
+  // When discussed with other devs, the concensus was to have a stateful
+  // component to house our checkboxes' state. But all other state is in Redux.
+  // Is there an easy way to house individual checkbox state in Redux?
+
+  componentDidMount() {
+    focusElement('.va-nav-breadcrumbs-list');
+    this.props.getDLCBatteryData();
+    // NOTE: Decision was to take socks out of MVP -@maharielrosario at 1/28/2020, 9:49:20 AM
+    // this.props.getDLCSocksData();
+    this.props.getDLCAccessoriesData();
+  }
+
+  createDLCUIData = dlcData => {
+    const dlcUIData = dlcData.map(dlcDataObj => {
+      let checkbox;
+      if (dlcDataObj.productGroup === 'ACCESSORIES') {
+        checkbox = (
+          <ErrorableCheckbox
+            className="vads-u-margin-left--4"
+            onValueChange={value => value}
+            id={dlcDataObj.id}
+          />
+        );
+      } else {
+        checkbox = (
+          <ErrorableCheckbox
+            className="vads-u-margin-left--2p5"
+            onValueChange={value => value}
+            id={dlcDataObj.id}
+          />
+        );
+      }
+      return {
+        ...dlcDataObj,
+        checkbox,
+      };
+    });
+    return dlcUIData;
+  };
+
+  render() {
+    const {
+      firstName,
+      lastName,
+      dlcBatteryData,
+      // NOTE: Decision was to take socks out of MVP -@maharielrosario at 1/28/2020, 9:49:20 AM
+      // dlcSocksData,
+      dlcAccessoriesData,
+      address,
+      city,
+      state,
+      zip,
+    } = this.props;
+
+    // NOTE: Only run the following code when the DLC's dlcData
+    // has been received to improve performance
+    let dlcUIBatteryData = [];
+    if (dlcBatteryData.length > 0) {
+      const returnedData = this.createDLCUIData(dlcBatteryData);
+      dlcUIBatteryData = returnedData;
+    }
+    // NOTE: Decision was to take socks out of MVP -@maharielrosario at 1/28/2020, 9:49:20 AM
+    // let dlcUISocksData = [];
+    // if (dlcSocksData.length > 0) {
+    //   const returnedData = this.createDLCUIData(dlcSocksData);
+    //   dlcUISocksData = returnedData;
+    // }
+
+    let dlcUIAccessoriesData = [];
```

Comment:
we could probably do a better job with semantically naming our variables throughout this work.  If I were to work in this code, I wouldn't really know what dlcUIData is.

Reply:
This is been deleted.